### DISCUSSION TITLE
Add support for teleports to work across bungee

### DISCUSSION THREAD

**Discussion by WarWolf-1** [Zrips/CMI#5543] [Enhancement]
Add support for teleports to work across the bungee network. For an example, players should be able to /tpa to a player in another server in a bungee network

**Comment by CloudeLecaw** [comment]
Can be done by portals at the moment

**Comment by WarWolf-1** [comment]
But what about /tpa,/tpo,/tpahere etc?

**Comment by CloudeLecaw** [comment]
Why should it? Other player dont see you so tpa etc is not usefull here
Its just usefull if they play on same server then tpa and co works.

**Comment by WarWolf-1** [comment]
Lets put it this way. Can such a feature be added what allows the player to see a tpa request across the whole bungee network, accept it and have the player being teleported to you from a different server?

**Comment by CloudeLecaw** [comment]
May everything can be added if the developer sees a usefull Situation where thisnmakes sense.
Let Zrips decide if he wanne add tp request to servers that are just connected by a Bungee without players play together on same server...

**Comment by WarWolf-1** [comment]
Currently player has to switch servers with /[server] and then /tpa [player]

**Comment by CloudeLecaw** [comment]
Correct and thats what all servers do there is a reason for this ;)
Its ressourey enough already to creat the bunggee tabsbabd Chats that let them chat in one bunggee

**Comment by WarWolf-1** [comment]
Well, that is a bad argument against this

**Comment by CloudeLecaw** [comment]
There us no technical reason that makes sense todo this.
Even if two pplnwanne play together they neednto be on one same server and this is counting for bungees everywhere. Thats how it was made.
There are allready Plugins that trie this BungecordTeleport thing ..and yes.. they are all not really good and mosty send to much useless packages between the bungees.

**Comment by WarWolf-1** [comment]
It shouldn't even be hard to do it. Player 1 sends player 2 a tp request. listen for accept/decline. if it is accepted, send player 1 to the server player 2 is in and teleport player 1 to player 2 location.

**Comment by CloudeLecaw** [comment]
Not possible
You descripe the technical theroie of tp on a single server.
This does notnwork for bungee thaat way.

But hey. You may can develop a "simple" plugin ..you Said "its ni big thing" butbthe fact is
It is a big thing that needs tons of extra api code and pakats that cmi does not have now

**Comment by WarWolf-1** [comment]
Which part of it is not possible? There is a plugin that can do that, but I am trying to avoid extra plugins.

**Comment by CloudeLecaw** [comment]
Ok you creat the API for it. Then i create the tp command for. You said its possible and not so hard. You cab do it.  Over and out :P

**Comment by WarWolf-1** [comment]
I don't even have the source code for cmi. Lets see what the dev has to say about this

**Comment by CloudeLecaw** [comment]
Use cmis api, its free source here on git

The tool you need is called
In Class Translator 1.1

**Comment by WarWolf-1** [comment]
Interesting. I could take a look at it, but since I have 0 experience with it, I can't give any promises.

**Comment by CloudeLecaw** [comment]
https://github.com/Zrips/CMI-API
Here u have all u need

And here cmis api wiki
https://www.zrips.net/cmi/api/

Good luck :)
Tell me when the api is done so i can start with the command ;)

**Comment by CloudeLecaw** [comment]
No Problem that is why you write it. U said its easy

**Comment by WarWolf-1** [comment]
I'll see what I can do about it.

**Comment by CloudeLecaw** [comment]
Ok lets finish this here. U may see its more fun parts then real meant stuff.

Final fact is your requests is not possible to develoo with the code that cmi and bungee offers at the moment. This would cost to much time to develop even as you as enduser thinks its just alittle Feature.

That the Problem with requests, often the users thinks "hey its just a small thing" but to write it is a big one that mostly does not worth the affort.

**Comment by WarWolf-1** [comment]
Why even let players make requests when pretty much anything is too much?

**Comment by CloudeLecaw** [comment]
Lol to tell if it is possible or not. Request may look like a wishbox, butbaccept the limits of minecrafts code ;)

**Comment by WarWolf-1** [comment]
You really expect random people to know the limits? Just because the plugin does not have it built-in does not mean it's not within the limits of what's possible

**Comment by CloudeLecaw** [comment]
funny man^^ i love you xD
